puma_motor_driver/include/puma_motor_driver/driver.h

```diff
@@ -89,19 +88,20 @@ class Driver
   void setEncoderCPR(uint16_t encoder_cpr);
   void setGearRatio(float gear_ratio);
   void setMode(uint8_t mode);
-  void setMode(uint8_t mode, float p, float i, float d);
-  void setGains(float p, float i, float d);
+  void setMode(uint8_t mode, double p, double i, double d);
+  void setGains(double p, double i, double d);
 
+  uint8_t lastFault();
+  uint8_t lastPower();
+  uint8_t lastMode();
   float lastDutyCycle();
   float lastBusVoltage();
   float lastCurrent();
-  float lastTemperature();
-  float lastPosition();
-  float lastSpeed();
-  uint8_t lastFault();
-  uint8_t lastPower();
   float lastOutVoltage();
-  uint8_t lastMode();
+  float lastTemperature();
+  double lastPosition();
+  double lastSpeed();
+  double lastSetpoint();
```

Comment:
Can we get a comment block over each of these blocks of functions explaining what's going on with them? Eg, what do the `set*` functions, and what's the relationship between the `request*` and `last*` sets of functions?